services/shuttle-poise/src/lib.rs

```diff
@@ -12,6 +23,16 @@ where
     E: Send + Sync + 'static,
 {
     async fn bind(mut self, _addr: SocketAddr) -> Result<(), shuttle_runtime::Error> {
+        #[cfg(feature = "poise")]
+        serenity::ClientBuilder::new(self.token, self.intents)
+            .framework(self.framework)
+            .await
+            .map_err(shuttle_runtime::CustomError::new)?
+            .start_autosharded()
+            .await
+            .map_err(shuttle_runtime::CustomError::new)?;
```

Comment:
When poise 0.5 has aged, we can consider having the poise integration just be a part of shuttle-serenity. If I'm not mistaken, you would just make a serenity client, add the poise framework, and then use it just like plain shuttle-serenity.

Reply:
I didn't realise this but yeah, it totally should! Would be nice to significantly clean things up.

Reply:
Hmm... why wait for poise 0.5 to age? We can simply deprecate shuttle-poise in the state before this PR, but with a deprecation warning and a small example on how to use shuttle-serenity + 0.12 + poise 0.6. Does this sound good @oddgrd ?

Reply:
Sounds great to me!